Add contexts to "Sale" translatable strings

mlocati opened the pull request:
I added a context and a note for translators in all the translatable strings that contain "sale": that way it will be clear if "sale" refers to discounting or to selling.

Of course, a review by a native speaking developer is much appreciated (guess who am I talking about? :wink:)

Close #795

mlocati commented:
@Mesuva is there any problem with this PR?

mlocati commented:
> [...] those should be above [...]
> [...] based on my feedback [...]

I'm sorry, but I don't know to which feedback you are referring to...

mlocati commented:
Maybe you started the review in GitHub but didn't "publish" your notes? I don't see any...

Mesuva commented:
Oh weird, try that now. I reckon github is a bit different to the last time I used the 'review' feature.

mlocati commented:
> Oh weird, try that now. I reckon github is a bit different to the last time I used the 'review' feature.

Yep, now I see them!

mlocati commented:
I've rebased this PR and added https://github.com/concretecms-community-store/community_store/pull/803/commits/40b5bcac5d798a3f8b598be882e314e9c41cde1f

mlocati commented:
@Mesuva what about this PR?

Mesuva commented:
I must have just missed this one, sorry!

mlocati commented:
No problem! And thank you!!
